### YamlMime:PythonClass
uid: azure.mgmt.dataprotection.operations.DataProtectionOperations
name: DataProtectionOperations
fullName: azure.mgmt.dataprotection.operations.DataProtectionOperations
module: azure.mgmt.dataprotection.operations
summary: "> [!WARNING]\n> DO NOT instantiate this class directly.\n>\n> \n>\n> Instead,\
  \ you should access the following operations through\n>\n> <xref:azure.mgmt.dataprotection.DataProtectionMgmtClient>'s\n\
  >\n> <xref:data_protection> attribute.\n>"
constructor:
  syntax: DataProtectionOperations(*args, **kwargs)
methods:
- uid: azure.mgmt.dataprotection.operations.DataProtectionOperations.check_feature_support
  name: check_feature_support
  summary: 'Validates if a feature is supported.


    Validates if a feature is supported.'
  signature: 'check_feature_support(location: str, parameters: _models.FeatureValidationRequestBase,
    *, content_type: str = ''application/json'', **kwargs: Any) -> _models.FeatureValidationResponseBase'
  parameters:
  - name: location
    description: Required.
    isRequired: true
    types:
    - <xref:str>
  - name: parameters
    description: 'Feature support request object. Is either a FeatureValidationRequestBase

      type or a IO[bytes] type. Required.'
    isRequired: true
    types:
    - <xref:azure.mgmt.dataprotection.models.FeatureValidationRequestBase>
    - <xref:typing.IO>[<xref:bytes>]
  return:
    description: FeatureValidationResponseBase or the result of cls(response)
    types:
    - <xref:azure.mgmt.dataprotection.models.FeatureValidationResponseBase>
  exceptions:
  - type: azure.core.exceptions.HttpResponseError
attributes:
- uid: azure.mgmt.dataprotection.operations.DataProtectionOperations.models
  name: models
  signature: models = <module 'azure.mgmt.dataprotection.models' from 'C:\\ToolCache\\Python\\3.11.9\\x64\\Lib\\site-packages\\py2docfx\\venv\\venv110\\Lib\\site-packages\\azure\\mgmt\\dataprotection\\models\\__init__.py'>
